### ISSUE TITLE
`block_comment_001` and `block_comment_003` fails without `--fix`, after fixed.

### ISSUE THREAD

**Issue by bennyslbs** [jeremiah-c-leary/vhdl-style-guide#1571]
**Environment**
OS: PRETTY_NAME="Linux Mint 22.3"
vsg --version: VHDL Style Guide (VSG) version: 3.35.0.post178+git.4aef7632
vsg branch: master

**Describe the bug**
`block_comment_001` and `block_comment_003` fixes as wanted when vsg --fix, but gives an error without --fix, the error is:
```
--------------------+------------+------------+--------------------------------------
  Rule              |  severity  |  line(s)   | Solution
--------------------+------------+------------+--------------------------------------
  block_comment_001 | Error      |          5 | Change block comment header to : ------------------------------------------------------------------------------------------------------------------------
  block_comment_003 | Error      |          7 | Change block comment footer to : ------------------------------------------------------------------------------------------------------------------------
--------------------+------------+------------+--------------------------------------
```

I want the comment indented to the same level as the code is, and `--fix` does that.

**To Reproduce**
1. Run `--c vsg_config.yaml.txt -f ex.vhd.txt --fix`
2. Verify no warnings or errors are reported
3. Run `--c vsg_config.yaml.txt -f ex.vhd.txt`
4. See errors reported - as shown above.

**Expected behavior**
The verify (no `--fix`) don't report errors on code that have been `--fix`ed.

**Screenshots**
None

**Additional context**
See attached minimal `vsg_config.yaml`.txt + `ex.vhd`.txt.

[vsg_config.yaml.txt](https://github.com/user-attachments/files/30655450/vsg_config.yaml.txt)

[ex.vhd.txt](https://github.com/user-attachments/files/30655463/ex.vhd.txt)

**Comment by JHertz5** [comment]
Hi @bennyslbs.

Thanks for raising this issue. I can recreate the issue and agree that the behaviour is problematic. I have a couple of questions:

1. The example config includes `library_008`. This rule shouldn't have any impact. Was this included by accident?
2. It looks to me like the issue only occurs when the architecture is empty (apart from comments obviously). Does that match your experience as well?

## Comments only
HDL
```vhdl
architecture rtl of ex is

begin

  ----------------------------------------------------------------------------------------------------------------------
  -- Lorem ipsum
  ----------------------------------------------------------------------------------------------------------------------

  -- x <= y;

end architecture rtl;
```
ouptut
```
$ bin/vsg -f test.vhd -c test.yml
================================================================================
File:  test.vhd
================================================================================
Phase 1 of 7... Reporting
Total Rules Checked: 2
Total Violations:    2
  Error   :     2
  Warning :     0
--------------------+------------+------------+--------------------------------------
  Rule              |  severity  |  line(s)   | Solution
--------------------+------------+------------+--------------------------------------
  block_comment_001 | Error      |          5 | Change block comment header to : ------------------------------------------------------------------------------------------------------------------------
  block_comment_003 | Error      |          7 | Change block comment footer to : ------------------------------------------------------------------------------------------------------------------------
--------------------+------------+------------+--------------------------------------
NOTE: Refer to online documentation at https://vhdl-style-guide.readthedocs.io/en/latest/index.html for more information.
```

## Not only comments
HDL
```vhdl
architecture rtl of ex is

begin

  ----------------------------------------------------------------------------------------------------------------------
  -- Lorem ipsum
  ----------------------------------------------------------------------------------------------------------------------

  x <= y;

end architecture rtl;
```
Output
```
$ bin/vsg -f test.vhd -c test.yml
================================================================================
File:  test.vhd
================================================================================
Phase 7 of 7... Reporting
Total Rules Checked: 2
Total Violations:    0
  Error   :     0
  Warning :     0
```

I have a theory about what could be causing this, and I fear that the solution might be quite complex, but I'll keep investigating.

**Comment by JHertz5** [comment]
OK, I've worked out part of why this is the case. The function `set_indent_of_normal_comment` uses `get_indent_value_of_next_token` to set the indent of the comment. This is very useful in 99% of cases, because you almost always want to have the comment aligned with the thing after it, e.g.
```vhdl
architecture rtl of ex is begin
  -- Comment
  x <= y;
end architecture rtl;
```
rather than
```vhdl
architecture rtl of ex is begin
-- Comment
  x <= y;
end architecture rtl;

```

So the question remains - why does the behaviour differ between the checking and fixing? It seems that checking expects the indentation to be removed, so the comment needs to be longer, while fixing expects the indentation to remain. I'll keep digging.

**Comment by bennyslbs** [comment]
> * The example config includes `library_008`. This rule shouldn't have any impact. Was this included by accident?
That was by accident, In the beginning I thought  it had something to do with our indention of `    use ieee.std_...`.
  
> * It looks to me like the issue only occurs when the architecture is empty (apart from comments obviously). Does that match your experience as well?
No, I observed the issue in real acitectures as well:
- In architecture in declaretive part
  - Exception: If a comment line exists just before the block comment, then OK.
    - If a blank line or code line after the comment, then the error occours.
    - If multiple block comments with a blank line in between, then there must be a comment just before each block comment to let the block comment be OK.
- In architecture after begin:
  - No error before code and after processes, but after instantiation the error occours.
    - A code (process or concurrent) statement after instantiation does not remove the error
    - Exception:
      - As for declaretive part
      - If there are code after instantiation, then the block comments betweeen instantiation and code are OK.
        - A comment instead of code does not make the block comment OK.

The explanation of when it occours gets messy.
Tell me if you need more examples and if I shall make a file with a lot of architectures that fails and if it also is required with a file with "fixes" like comments or code that hides the issue to let the comment be OK.
